compaction/compaction_manager.hh

```diff
@@ -40,6 +40,7 @@ class compacting_sstable_registration;
 // behalf of existing tables.
 class compaction_manager {
 public:
+    using compaction_result_counters = std::optional<sstables::compaction_counters>;
```

Comment:
I have doubts about this shortname.
It lets a reader think there is yet another struct holding counters that is different than `compaction_counters` itself.

In other places we use a convention for a `_opt` suffix for optional typedefs, so in this case it would be
`using compaction_counters_opt = std::optional<sstables::compaction_counters>;`
This better describes the type.

For example, I didn't figure out why in your map_reduce implementation you instatiated an optional for the initial value, until I got here...